common/service/pldm/pldm_firmware_update.c

```diff
@@ -548,8 +592,15 @@ static uint8_t pass_component_table(void *mctp_inst, uint8_t *buf, uint16_t len,
 		goto exit;
 	}
 
-	if (current_state == STATE_IDLE) {
-		resp_p->completion_code = PLDM_FW_UPDATE_CC_NOT_IN_UPDATE_MODE;
+	LOG_INF("Received component class: %xh id: %d with version:", req_p->comp_classification,
+		req_p->comp_identifier);
+	LOG_HEXDUMP_INF(buf + sizeof(struct pldm_pass_component_table_req), req_p->comp_ver_str_len,
+			"");
+
+	if (current_state != STATE_LEARN_COMP) {
+		LOG_ERR("Current mode %d doesn't meet request state %d", current_state,
```

Comment:
Similar to above, this should indicate something like "firmware update failed because current state %d is invalid", current_state